handlers/DataSourcesController.go

```diff
@@ -89,10 +90,12 @@ var CreateDataSourceHandler = func() func(c *gin.Context) {
 			logrus.Errorf("Unable to get bytes from request. ", err)
 			c.Status(http.StatusBadRequest)
 		} else {
-			id, err := service.CreateDataSource(c.Request.Context(), bytes)
-			if err != nil || id == nil {
+			id, err := service.CreateDataSource(c.Request.Context(), bytes, c.Param("org"))
+			if err != nil && strings.Contains(err.Error(), "Bad Request") {
+				c.JSON(http.StatusBadRequest, err.Error())
+			} else if err != nil || id == nil {
```

Comment:
Hvis du flytter mer av error-håndtering og loggingen nærmere der feilen skjer til CreateDataSource hvor det allerede er en del håndtering av feil, så kan controller gjøres enklere. Optimalt at controller selv ikke kjenner detaljene om feil.

Reply:
Bruker det mest her for å bestemme type respons/status, men kan sikkert få det til på en bedre måte. Skal tenke litt